src/material-experimental/mat-feature-highlight/feature-highlight-container.ts

```diff
@@ -0,0 +1,533 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directionality} from '@angular/cdk/bidi';
+import {coerceCssPixelValue} from '@angular/cdk/coercion';
+import {ComponentPortal, TemplatePortal} from '@angular/cdk/portal';
+import {AfterViewInit, ChangeDetectorRef, Component, ComponentRef, ElementRef, EmbeddedViewRef, EventEmitter, OnDestroy, Output, ViewChild} from '@angular/core';
+
+import {FeatureHighlightCalloutContainer} from './feature-highlight-callout-container';
+import {FeatureHighlightConfig} from './feature-highlight-config';
+
+interface Coordinate {
+  x: number;
+  y: number;
+}
+
+type State = 'none'|'active_with_animation'|'active_without_animation'|
+    'fade_accept'|'fade_dismiss';
+
+const OPEN_ANIMATION_TIME_MS = 350;
+const CLOSE_ANIMATION_TIME_MS = 200;
+
+const DISMISS_EVENT = 'click';
+
+/**
+ * Internal component that holds feature highlight elements as well as
+ * user-provided feature highlight content.hostelement
+ */
+@Component({
+  selector: 'feature-highlight-container',
+  templateUrl: './feature-highlight-container.ng.html',
+  styleUrls: ['./feature-highlight-container.css'],
+  host: {
+    '[attr.aria-describedby]': 'config.ariaDescribedBy',
+    '[attr.aria-label]': 'config.ariaLabel',
+    '[attr.aria-labelledby]': 'config.ariaLabel ? null : config.ariaLabelledBy',
+    '[attr.role]': '"alertdialog"',
+    '[class.fade-accept]': 'state === "fade_accept"',
+    '[class.fade-dismiss]': 'state === "fade_dismiss"',
```

Comment:
These class names are very generic and have a chance to conflict with classes from the consumer. They should be prefixed with `mat-`.